### PR TITLE
Implement Subtitle Property for ListViewGroup

### LINKED ISSUE #2655: Add support for subtitles in ListViewGroup
- You can set the `LVGF_SUBTITLE` flag on `mask` and set `pszSubtitle` to a valid string and `cchSubtitle` to the length of the string

## Proposed API
This would add the following API to `ListViewGroup`:
```cs
public class ListViewGroup
{
    ...
    public string Subtitle { get; set; }
    ...
}
```

# Example
```cs
var group = new ListViewGroup
{
     Subtitle = "Subtitle"
};
```

E.g.
image

### PR DESCRIPTION
**Pull request by lonitra**
Fixes #2655

## Proposed changes

- Add subtitle API to `ListViewGroup`

## Customer Impact

- Customers will have the ability to add subtitles to their ListViewGroups

## Regression? 

- No

### Before

image

### After

image

## Test methodology 

- Unit testing
- Manual testing

## Note
- Correct me if I'm wrong, but for accessibility purposes I believe we'll need to expose the UIA Text Pattern

###### Microsoft Reviewers: [Open in CodeFlow](http://wpcp.azurewebsites.net/CodeFlowProtocolProxy2.php?pullrequest=https://github.com/dotnet/winforms/pull/3234)

### CODE REVIEW

**Comment by RussKie** [comment]
Merge conflict.

Also did you run the skipped tests with the RemoteExecutor locally?

File: src/System.Windows.Forms/src/System/Windows/Forms/ListViewGroup.cs [hunk 1]
```diff
@@ -160,6 +161,25 @@ public HorizontalAlignment FooterAlignment
             }
         }
 
+        /// <summary>
+        ///  The text displayed in the group subtitle.
+        /// </summary>
+        [SRCategory(nameof(SR.CatAppearance))]
+        public string Subtitle
```
Comment (hughbe) [inline src/System.Windows.Forms/src/System/Windows/Forms/ListViewGroup.cs:168]: Should have `AllowNull` attribute to signify that we allow it to be set to null but it doesn't return null

**Comment by hughbe** [comment]
Also, MC

**Comment by lonitra** [comment]
> Also did you run the skipped tests with the RemoteExecutor locally?
Yes I ran these locally without the skip statement and they all passed

**Comment by RussKie** [comment]
Have you had a chance to run accessibility assessment on this?

**Comment by lonitra** [comment]
> Have you had a chance to run accessibility assessment on this?

Both inspect and narrator don't pick up on the subtitle unfortunately. I'm thinking we would need to expose the [UIA Text Pattern](https://docs.microsoft.com/en-us/dotnet/framework/ui-automation/ui-automation-textpattern-overview) or override [AccessibleObject value property,](https://docs.microsoft.com/en-us/dotnet/api/system.windows.forms.accessibleobject.value?view=netcore-3.1) but I'm not sure which, if either, is the right way to go here.

**Comment by RussKie** [comment]
Please reach out to @m-lipin for directions

**Comment by M-Lipin** [comment]
Accessibility for ListViewGroup Subtitle announcement will be covered in #3224.

**Comment by RussKie** [comment]
Thank you

**Pull request merged**